lib/entities/job.tests.js

```
@@ -0,0 +1,73 @@
+import expect, { spyOn } from 'expect';
+import Dispatch from '../dispatch';
+import nock from 'nock';
+import * as endpoints from '../endpoints'
+const testClientID = '12345';
+const testClientSecret = '54321';
+const testBearerToken = 'this is the access token';
+const testRefreshToken = 'this is the refresh token';
+
+describe('job', () => {
+  describe('getNotes', () => {
+    it('should call endpoint with correct query string', () => {
+      const client = new Dispatch(testClientID, testClientSecret);
+      client.setBearerToken(testBearerToken, testRefreshToken);
+      const scope = nock('https://api.dispatch.me')
+        .get(`${endpoints.ATTACHMENTS}?filter[file_token_null]=1&filter[entity_type]=Job&filter[entity_id]=123`)
+        .reply(200);
+      client.entities.job(123).getNotes();
+      expect(scope.isDone()).toEqual(true);
+    });
+
+    it('should parse response correctly', done => {
+      const client = new Dispatch(testClientID, testClientSecret);
+      client.setBearerToken(testBearerToken, testRefreshToken);
+      const scope = nock('https://api.dispatch.me')
+        .get(`${endpoints.ATTACHMENTS}?filter[file_token_null]=1&filter[entity_type]=Job&filter[entity_id]=123`)
+        .reply(200, {
+          attachments:[{
+            id: 1,
+            file_token: null,
+            entity_type: 'Job',
+            entity_id: 123,
+          }],
+        });
+      client.entities.job(123).getNotes().then(notes => {
+        expect(notes.length).toEqual(1);
+        done();
```

Aha, I see. I didn't realize it equated to `done(err)` in the catch block. I was worried it would silently pass on failures. I'm not sure, but I doubt it behaves differently for `chai`.